internal/wire/record.go

```diff
@@ -56,28 +137,68 @@ func LoadRecords(dir string, limit int) ([]Record, error) {
 		return nil, nil
 	case err != nil:
 		return nil, lazyerrors.Error(err)
+	default:
+		return files, nil
 	}
+}
 
-	if limit > 0 && len(files) > limit {
-		f := make([]string, limit)
-		for fI, filesI := range rand.Perm(len(files))[:limit] {
-			f[fI] = files[filesI]
+// extractRecordDataDocuments collects all data documents from a given record.
+func extractRecordDataDocuments(r Record) ([]*types.Document, error) {
+	var docs []*types.Document
+
+	switch b := r.Body.(type) {
+	case *OpMsg:
+		doc, err := b.Document()
+		if err != nil {
+			return nil, err
 		}
-		files = f
+		docs, err = appendDataDocuments(docs, doc)
+		if err != nil {
+			return nil, lazyerrors.Errorf("OpMsg: %w", err)
+		}
+
+	case *OpReply:
+		var err error
+		for i, d := range b.Documents {
+			docs, err = appendDataDocuments(docs, d)
+			if err != nil {
+				return nil, lazyerrors.Errorf("OpReply.documents[%d]: %w", i, err)
+			}
+		}
+		docs = append(docs, b.Documents...)
 	}
 
-	var res []Record
+	return docs, nil
+}
 
-	for _, file := range files {
-		r, err := loadRecordFile(file)
-		if err != nil {
-			return nil, lazyerrors.Errorf("%s: %w", file, err)
+// appendDataDocuments finds all valid data documents in v and pushes them to dst.
+func appendDataDocuments(dst []*types.Document, v any) ([]*types.Document, error) {
+	var err error
+
+	switch v := v.(type) {
+	case *types.Document:
+		if v.ValidateData() == nil {
+			dst = append(dst, v)
+			return dst, nil
 		}
 
-		res = append(res, r...)
+		if v.Has("insert") && v.Has("documents") {
```

Comment:
This one looks like a record from insert command. Does fuzzing require or forbid having `_id` field? I'm not familiar with stored records but I suppose it may or may not have `_id` field :thinking: